_docs/pql_queries.md

```diff
@@ -116,6 +116,29 @@ puppet query 'catalogs[certname,producer_timestamp] {  producer_timestamp < "202
 puppet query 'reports[certname,transaction_uuid,receive_time] { cached_catalog_status="on_failure" and start_time > "2021-10-27T15:36:00-05:00" and end_time < "2021-10-27T16:35:00-05:00"  }'
 ```
 
+### Get all nodes with corrective changes
+
+It heavily improves the performance when we filter for events from the past
+hours. PuppetDB uses [partitioning](https://www.postgresql.org/docs/current/ddl-partitioning.html)
+and saves reports on a per-day partition. By filtering for the current day we
+avoid scans on other partitions.
+
+```
+puppet query 'events[certname]{corrective_change = true and report_receive_time > "2024-04-05T06:06:00.000Z" group by certname}'
```

Comment:
maybe replace the date with `$(date -u --date='1 day ago' +%Y-%m-%dT%H:%M:%S.000Z)` so it always gets just certs within the last day